src/main/kotlin/org/wfanet/measurement/kingdom/service/api/v2alpha/ExchangeStepAttemptsService.kt

```diff
@@ -52,3 +76,42 @@ class ExchangeStepAttemptsService : ExchangeStepAttemptsCoroutineImplBase() {
     TODO("world-federation-of-advertisers/cross-media-measurement#3: implement this")
   }
 }
+
+private fun InternalExchangeStepAttempt.toV2Alpha(): ExchangeStepAttempt {
+  return ExchangeStepAttempt.newBuilder().also { builder ->
+    builder.key = toV2AlphaKey()
+    builder.attemptNumber = attemptNumber
+    builder.state = state.toV2Alpha()
+    builder.addAllDebugLogEntries(details.debugLogEntriesList.map { it.toV2Alpha() })
+    builder.addAllSharedOutputs(details.sharedOutputsList)
+    builder.startTime = details.startTime
+    builder.updateTime = details.updateTime
+  }.build()
+}
+
+private fun InternalExchangeStepAttempt.State.toV2Alpha(): ExchangeStepAttempt.State {
+  return when (this) {
+    InternalExchangeStepAttempt.State.STATE_UNSPECIFIED,
+    InternalExchangeStepAttempt.State.UNRECOGNIZED ->
+      failGrpc(Status.INTERNAL) { "Invalid State: $this" }
+    InternalExchangeStepAttempt.State.ACTIVE -> ExchangeStepAttempt.State.ACTIVE
+    InternalExchangeStepAttempt.State.SUCCEEDED -> ExchangeStepAttempt.State.SUCCEEDED
+    InternalExchangeStepAttempt.State.FAILED -> ExchangeStepAttempt.State.FAILED
+  }
```

Comment:
`else -> failGrpc(Status.INTERNAL) { "Unrecognized State: $this" }`

Reply:
So, for this one, we can force a compilation error by not exhaustively handling the cases. It will look like:

```
src/main/kotlin/org/wfanet/measurement/kingdom/service/api/v2alpha/ExchangeStepAttemptsService.kt:93:10: error: 'when' expression must be exhaustive, add necessary 'SOME_NEW_CASE' branch or 'else' branch instead
```

Since the enum is "owned" -- that is, it's in the same repository and not controlled by a different party, it's safer (in my opinion) to force the compilation error and make the programmer handle the case, rather than just have it silently pass all tests and then blow up once deployed.

Does that make sense or would you rather have an `else` still? Note that since `UNRECOGNIZED` is not handled, we don't have to worry about problems caused by version skew -- when deployed, any number not known to be `ACTIVE`, `SUCCEEDED`, or `FAILED` will become `UNRECOGNIZED`.

Reply:
So, I agree that in its current state, other choices would force a compilation error. But, I do think it would be good to assert or aggressively log it just in case here. What do you think?

Reply:
Ironically, if we handle the case, we will prevent the compilation errors that we want, which will make it harder to debug. It's been our convention to be pretty deliberate about leveraging the compiler to prevent unhandled cases, e.g. here: https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/main/src/main/kotlin/org/wfanet/measurement/duchy/db/computation/ComputationsServiceCoroutineStubs.kt#L64.

I'd strongly prefer to leave this as is.